.changeset/slimy-starfishes-report.md

```diff
@@ -0,0 +1,6 @@
+---
+'svelte-ux': patch
+---
+
+Added new `IconInput` and `IconData` types to enable inclusive & seamless passing of icon arguments between components. Also provides a `asIconData` utility function for type-safe conversion.
+Fixed type errors for Button & TextField's use of Icon data.
```

Comment:
`<Icon>` used to just accept `<Icon path="...">`

...then it was extended to allow `<Icon svg="...">`, ``<Icon svgUrl="http://...">`

...then it was extended to have a simple `<Icon data="...">` prop and infer which of the props was intended (path, svg, svgUrl), so I call that a feature to allow `TextField` to take in more options than just `path` :).

`svgUrl` is quite useful when your loading over the network, and not populating your bundle with all potential options.

Reply:
Hehe, I wasn't saying that it doesn't support different kinds of inputs. More that, we have instances of wrappers with inner wrappers with inner-inner Icons, etc. and the allowed data types for those wrapper components' distinct arguments do not always match, leading to inconsistency that has the potential to force type errors for no reason. I actually have to compliment this project for the diversity of data formats that `Icon` supports.

Reply:
:).  It's been battled tested in a lot of use cases.  One thing I struggle with is getting the time to show all the niceties in examples :).  `<Table>` is under represented for how much it packs ;)